tests/unit/backend/layers/api/test_curation_api.py

```diff
@@ -961,18 +961,29 @@ def test__delete_dataset(self):
         ]
         for auth, auth_description, expected_status_code in auth_credentials:
             with self.subTest(f"{auth_description} {expected_status_code}"):
-                dataset = self.generate_dataset(
-                    statuses=[DatasetStatusUpdate(DatasetStatusKey.UPLOAD, DatasetUploadStatus.UPLOADING)],
-                    publish=False,
-                )
-
-                test_url = (
-                    f"/curation/v1/collections/{dataset.collection_version_id}/datasets/"
-                    f"{dataset.dataset_version_id}"
-                )
-                headers = auth() if callable(auth) else auth
-                response = self.app.delete(test_url, headers=headers)
-                self.assertEqual(expected_status_code, response.status_code)
+                with self.subTest("with version_ids"):
```

Comment:
nested subtests are not supported by unittest or pytest You'll need to add these details to the out subtest. That should also reduce the code duplication

Reply:
It's not easy to add them to the main subtest since the ids cannot be determined before the dataset is generated, and I'd avoid to add an if. I just added a function to remove some of the duplication.